src/Assignments/ProblemIframe.tsx

```diff
@@ -51,7 +52,34 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
 
     const { setLastSavedAt, setLastSubmittedAt } = useCurrentProblemState();
 
+    useEffect(() => {
+        setProblem(problemProp);
+    }, [problemProp]);
+
     useEffect(()=>{
+        const fetchHTML = async () => {
+            const rendererHTML = await getHTML();
+            setProblem(problem => {
+                const matchString = `action="/backend-api/courses/question/${problem.id}\\?`;
+                const re = new RegExp(matchString);
+                const match = rendererHTML?.match(re);
+
+                // don't need to check if workbook ID is nil -- always is paired with appropriate problem object **MAINTAIN THIS EXPECTATION**
+                if (!_.isNil(rendererHTML) && 
+                    (!_.isNil(previewPath) || 
+                    !_.isNil(previewProblemSource) || 
+                    !_.isNil(match))
+                ) {
+                    // we match or else we're in a situation where there's nothing relevant to match *to*
+                    setRenderedHTML(rendererHTML);
+                } else {
+                    // do we want any notification of our state conflict or should we ditch this?
+                    if (!_.isNil(rendererHTML)) logger.warn('Someone has been rifling through the problems too quickly!');
```

Comment:
This is probably better as a debug since we turned on warning notifications. We will get messages about this.